internal/httpclient/models/logout_request.go

```diff
@@ -22,6 +22,9 @@ type LogoutRequest struct {
 
 	// RPInitiated is set to true if the request was initiated by a Relying Party (RP), also known as an OAuth 2.0 Client.
 	RpInitiated bool `json:"rp_initiated,omitempty"`
+	
+	// client, only set if rp_initiated
+	Client *OAuth2Client `json:"client,omitempty"`
```

Comment:
These files are auto-generated, you would need to add that to the actual type ( https://github.com/ory/hydra/blob/master/consent/types.go#L398 ) and of course it also needs to be set and stored!

Reply:
> These files are auto-generated, you would need to add that to the actual type ( https://github.com/ory/hydra/blob/master/consent/types.go#L398 ) and of course it also needs to be set and stored!

OK, the thing is, the client is actually passed, so ory hydra works with it, it just isnt represented in the json form, so I suppose with my latest commit it works now?

Reply:
But it already is set here https://github.com/ory/hydra/blob/24f91ab7038e4c612db5af86ff0f3ee5eb03c878/consent/strategy_default.go#L914

Reply:
Ah I see, my bad - these files are auto-generated! What you need to do is to change this line:

https://github.com/ory/hydra/blob/cf2fe0c999f3ef4971880439e5d04b2986fc890d/consent/types.go#L451

This will then update that file automatically! The we also need a test to check if the client is actually set in JSON. For example, by doing an assertion here: https://github.com/ory/hydra/blob/cf2fe0c999f3ef4971880439e5d04b2986fc890d/consent/strategy_logout_test.go#L140

Reply:
> Ah I see, my bad - these files are auto-generated! What you need to do is to change this line:
> 
> https://github.com/ory/hydra/blob/cf2fe0c999f3ef4971880439e5d04b2986fc890d/consent/types.go#L451
> 
> This will then update that file automatically! The we also need a test to check if the client is actually set in JSON. For example, by doing an assertion here:
> 
> https://github.com/ory/hydra/blob/cf2fe0c999f3ef4971880439e5d04b2986fc890d/consent/strategy_logout_test.go#L140

I already altered the types.go, but where should I put the test? There's also no test that epxects rp_inititated to be true, which is only case where client would be set.

Reply:
This test is doing rp_initiated: https://github.com/ory/hydra/blob/cf2fe0c999f3ef4971880439e5d04b2986fc890d/consent/strategy_logout_test.go#L343

You can add the assertion there!

Reply:
> This test is doing rp_initiated:
> 
> https://github.com/ory/hydra/blob/cf2fe0c999f3ef4971880439e5d04b2986fc890d/consent/strategy_logout_test.go#L343
> 
> You can add the assertion there!

Where exactly? I dont see an assert for rp_initiated, so Iam not sure.